server/aws-lsp-codewhisperer/src/shared/localProjectContextController.ts

```diff
@@ -138,11 +140,13 @@ export class LocalProjectContextController {
 
             // build index if vecLib was initialized but indexing was not enabled before
             if (this._vecLib) {
+                // if indexing is turned being on, build index with 'all' that supports vector indexing
                 if (enableIndexing && !this._isIndexingEnabled) {
```

Comment:
why are 2 booleans required

Reply:
this is to gate unnecessary update. It checks if the old value has changed and only then makes the buildIndex calls accordingly